src/pages/home.js

```diff
@@ -1,30 +1,62 @@
 import React, { useEffect } from 'react';
+import { Link } from 'react-router-dom';
 
 // Components
 import Post from '../components/post/postList';
 import Profile from '../components/profile';
+import NewPost from '../components/post/newPost';
 
 // MUI
 import Grid from '@material-ui/core/Grid';
+import Button from '@material-ui/core/Button';
+import withStyles from '@material-ui/core/styles/withStyles';
+import style from '../style/style';
+
+// Icons
+import AddIcon from '@material-ui/icons/Add';
 
 // Redux
 import { connect } from 'react-redux';
 import { getPostList } from '../redux/actions/dataActions';
 
-export const Home = ({ postList, isLoading, getPostList }) => {
+export const Home = ({
+  classes,
+  postList,
+  isLoading,
+  getPostList,
+  isAuthenticated,
+}) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   useEffect(() => getPostList(), []);
   let recentPostMarkup;
   if (!isLoading) {
     recentPostMarkup = postList.map(post => (
-      <Post key={post.postId} post={post} />
+      <Post key={`post-${post.postId}`} post={post} />
     ));
   } else {
     recentPostMarkup = <p>Loading...</p>;
   }
+  let createPostMarkup;
+  if (!isAuthenticated) {
+    createPostMarkup = (
```

Comment:
I'd suggest making createPostMarkup into "makePostCreateEditor" as a function:
```
const makeCreatePostEditor = () =>{
  if(isAuthenticated){
    return //stuff
  }
return //stuff
}
```
which you then use below as:
```
   {createPostMarkup()}
```

Reply:
Was not aware this was possible. Fixed.